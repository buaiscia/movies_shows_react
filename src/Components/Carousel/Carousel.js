import React, { Component } from 'react';
import { Redirect } from "react-router-dom";

import classes from './Carousel.module.css';

class Carousel extends Component {

    state = {
        redirect: null, //define the redirect state to check when to do it
    };

    handleClick = () => {
        // onclick image change redirect state 
        this.setState({ redirect: '/show' }, () => {
            if (this.props.onClick) {
                this.props.onClick(this.state);
            }
        })
    }


    render() {
        let props = this.props;

        // check if redirecting
        if (this.state.redirect) {
            let red = this.state.redirect;
            return <Redirect push to={{ //Connect to Route /show displaying Item component
                pathname: red, // passing Redirect pathname
                state: {       // passing different props to be used in Item component
                    id: props.id,
                    title: props.title,
                    name: props.name,
                    description: props.description,
                    poster: props.poster,
                    popularity: props.popularity,
                    vote: props.vote,
                }
            }
            } />
        }

        return (
            <>
                <div>
                    <img
                        className={classes.image}
                        onClick={this.handleClick} // call handleClick function
                        alt={`${props.title || props.name} poster`}
                        src={props.poster} />
                    <p className={classes.title}>{props.title || props.name}</p>
                </div>
            </>
        )
    }

}

export default Carousel;