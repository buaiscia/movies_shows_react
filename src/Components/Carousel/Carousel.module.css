.image {
    cursor: pointer;
    border-radius: 5%;
}

.title {
    padding-top: 2%;
}