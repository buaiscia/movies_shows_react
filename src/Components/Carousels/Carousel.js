import React, { Component } from 'react';
import { Redirect } from "react-router-dom";

class Carousel extends Component {


    state = { 
        redirect: null
    };

    handleClick = () => {
        this.setState({ redirect: '/movie'}, () => {
            if(this.props.onClick) {
                this.props.onClick(this.state);
            }
        })
    }
    

    render() {
        let props = this.props;
        
        
        if (this.state.redirect) {
            let red = this.state.redirect;
            return <Redirect to={{
                pathname : red,
                state : { 
                    id : props.id,
                    title : props.title,
                    name: props.name,
                    description : props.description,
                    poster: props.poster,
                    popularity: props.popularity,
                    vote: props.vote
                 }}
            }  />
          }

        return (
            <>
                <div>
                    <p>{props.title ? props.title : props.name}</p>
                    <img alt={`${props.title ? props.title : props.name} poster`} src={props.poster}  />
                    <button onClick={this.handleClick}>Click</button>
                    <p>{props.id}</p>
                </div>
            </>
        )
    }

}

export default Carousel;