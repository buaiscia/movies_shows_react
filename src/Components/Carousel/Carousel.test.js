import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import Carousel from './Carousel';

configure({ adapter: new Adapter() });


describe('Carousel', () => {
    it('should render correctly', () => {
        const component = shallow(
            <Carousel />
        );
        expect(shallowToJson(component)).toMatchSnapshot();
    });
    it('should have initial redirect state as null', () => {
        const wrapper = shallow(<Carousel />);
        expect(wrapper.state('redirect')).toEqual(null);
    });
    it('should change the redirect state when img clicked', () => {
        const wrapper = shallow(<Carousel />);
        wrapper.find("img").simulate('click', { redirect: { value: '/show' } });
        expect(wrapper.state('redirect')).toEqual('/show');
    });
}

)