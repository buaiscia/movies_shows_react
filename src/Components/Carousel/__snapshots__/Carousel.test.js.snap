// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel should render correctly 1`] = `
<Fragment>
  <div>
    <img
      alt="undefined poster"
      className="image"
      onClick={[Function]}
    />
    <p
      className="title"
    />
  </div>
</Fragment>
`;
